edu_edfi_airflow/dags/edfi_resource_dag.py

```diff
@@ -48,116 +63,155 @@ def __init__(self,
+        return configs
 
-        # Build an operator to increment the DBT var at the end of the run.
-        if self.dbt_incrementer_var:
-            self.dbt_var_increment_operator = self.build_dbt_var_increment_operator()
+    def parse_endpoint_configs(self, raw_configs: Optional[Union[dict, list]] = None) -> Tuple[Dict[str, dict], Set[str], Set[str]]:
+        """
+        Parse endpoint configs into kwarg bundles if passed.
+        Keep list of deletes and keyChanges to fetch.
+        Force all endpoints to snake-case for consistency.
+        Return only enabled configs (enabled by default).
+        """
+        if not raw_configs:
+            return {}, set(), set()
+        
+        # A dictionary of endpoints has been passed with run-metadata.
+        elif isinstance(raw_configs, dict):
+            configs = {}
+            deletes = set()
+            key_changes = set()
+
+            for endpoint, kwargs in raw_configs.items():
+                if not kwargs.get('enabled', True):
+                    continue
+
+                snake_endpoint = camel_to_snake(endpoint)
+                configs[snake_endpoint] = self.build_endpoint_configs(**kwargs)
+                if kwargs.get('fetch_deletes'):
+                    deletes.add(snake_endpoint)
+                    key_changes.add(snake_endpoint)
+            
+            return configs, deletes, key_changes
+
+        # A list of resources has been passed without run-metadata
+        elif isinstance(raw_configs, list):
+            # Use default configs and mark all as enabled with deletes/keyChanges.
+            configs = {camel_to_snake(endpoint): self.build_endpoint_configs(namespace=ns) for endpoint, ns in raw_configs}
+            deletes = set(configs.keys())
+            key_changes = set(configs.keys())
+            return configs, deletes, key_changes
+        
         else:
-            self.dbt_var_increment_operator = None
-
-        # Create nested task-groups for cleaner webserver UI
-        # (Make these lazy to only show populated TaskGroups in the UI.)
-        self.resources_task_group = None
-        self.resource_deletes_task_group = None
-        self.descriptors_task_group = None
-
-
-    def add_resource(self,
-        resource: str,
-        namespace: str = 'ed-fi',
-        **kwargs
-    ):
-        if not self.resources_task_group:  # Initialize the task group if still undefined.
-            self.resources_task_group = TaskGroup(
-                group_id="Ed-Fi Resources",
-                prefix_group_id=False,
-                parent_group=None,
-                dag=self.dag
+            raise ValueError(
+                f"Passed configs are an unknown datatype! Expected Dict[endpoint: metadata] or List[(namespace, endpoint)] but received {type(configs)}"
             )
+    
+    # Original methods to manually build task-groups (deprecated in favor of `resource_configs` and `descriptor_configs` DAG arguments).
```

Comment:
These methods will be removed in a future version of the code, but they were easy to retrofit to not break the existing DAG-initialization code.